packages/tailwindcss/src/utilities.ts

```diff
@@ -4854,8 +4854,35 @@ function resolveValueFunction(
       arg.value[0] === '-' &&
       arg.value[1] === '-'
     ) {
-      let resolved = designSystem.resolveThemeValue(arg.value.replace('*', value.value))
```

Comment:
`resolveThemeValue` is fully resolving the value because we passed it a value like `--text-sm--line-height`.

Instead, we should use the `resolveWith` in case of nested keys.